File: 0560.和为K的子数组/0560-和为K的子数组.c
int subarraySum(int* nums, int numsSize, int k){
    int len = numsSize + 1;
    int sums[len];
    for (int i=1; i<len;i++) {
        sums[i] = sums[i-1] + nums[i-1];
    }
    int ret = 0;
    for (int i=0;i<len;i++) {
        for (int j=i+1;j<len;j++) {
            if (sums[j] - sums[i] == k) {
                ret += 1;
            }
        }
    }
    return ret;
}

File: 0560.和为K的子数组/0560-和为K的子数组.rs
impl Solution {
    pub fn subarray_sum(nums: Vec<i32>, k: i32) -> i32 {
        let len = nums.len() + 1;
        let mut sums: Vec<i32> = Vec::with_capacity(len);
        sums.push(0);
        for i in 1..len {
            sums.push(sums[i-1] + nums[i-1]);
        }
        println!("{:?}",sums);
        let mut ret = 0;
        for i in 0..sums.len() {
            for j in (i+1)..sums.len() {
                if sums[j] - sums[i] == k {
                    ret += 1;
                }
            }
        }
        
        ret
    }
}